### ISSUE TITLE
ink! `4.0-beta` tracking

### ISSUE THREAD

**Issue by ascjones** [paritytech/ink#1478]
We agreed at the team [retreat](https://hackmd.io/eHhYA9UASeey5C8k-ZTIYQ) what needs to be done for [`ink! 4.0`](https://github.com/paritytech/ink/issues/1343) beta: essentially any contract ABI breaking changes MUST be included in this release. Breaking changes to the language itself (e.g. [shared events](https://github.com/paritytech/ink/pull/1243)) can be added in future beta pre-releases.

This first pre-release will be `beta`, followed by `beta.1`, `beta.2` etc.

## Must haves

- [x] https://github.com/paritytech/ink/pull/1450 @HCastano 
- [x] https://github.com/paritytech/ink/pull/1462 @agryaznov 
  - Blocked on updating `substrate-contract-node` to latest `polkadot` release, a follow up to https://github.com/paritytech/substrate-contracts-node/pull/156 @athei 
- [x] revert https://github.com/paritytech/ink/pull/1458 @SkymanOne 
- [x] https://github.com/paritytech/ink/pull/1463 @lean-apple 
- [x] https://github.com/paritytech/ink/issues/1483 @ascjones

## Nice to haves

- [x] https://github.com/paritytech/ink/pull/1460 @SkymanOne 
- [x] `cargo-contract` extrinsics compatibility https://github.com/paritytech/cargo-contract/pull/809 @ascjones 

## Other todos

- [ ] Prepare a release PR
